backend/contributions/tests/test_community_awards.py

```
@@ -425,6 +425,36 @@ def fail_on_second(*args, **kwargs):
+        response = self.confirm([{'user_id': self.member.id, 'amount': 300}])
+        self.assertEqual(response.data['status'], 'already_processed')
+        self.assertEqual(Contribution.objects.count(), 0)
+
+    def test_portal_revoke_is_atomic(self):
+        # A revoke log that outlives a failed delete is worse than no log: it
+        # records a removal that never happened.
+        self.confirm([{'user_id': self.member.id, 'amount': 300}])
+        award = Contribution.objects.get(contribution_type=self.trivia_type)
+
+        with patch.object(Contribution, 'delete', side_effect=RuntimeError('boom')):
+            with self.assertRaises(RuntimeError):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)